### PR TITLE
refactor(prov-device): Rename variables for accepting client certificate signing requests

### PR DESCRIPTION
**Pull request by abhipsaMisra**
Renamed `OperationalCertificateRequest` to `ClientCertificateSigningRequest`.
Also updated some doc comments to highlight that the signed certificate only contains the public key information.

### CODE REVIEW

**Review by brycewang-microsoft — APPROVED** [review]

File: common/src/device/provisioning/transport/DeviceRegistrationResult.cs [hunk 1]
```diff
@@ -68,9 +68,9 @@ internal partial class DeviceRegistrationResult
         public JRaw Payload { get; set; }
 
         /// <summary>
-        /// The PEM encoded operational client certificate that was returned by the certificate authority.
+        /// The PEM encoded operational client certificate that was signed by the certificate authority.
         /// This operational certificate was used by the device provisioning service to register the enrollment with IoT Hub.
```
Comment (timtay-microsoft) [inline common/src/device/provisioning/transport/DeviceRegistrationResult.cs:72]:
```suggestion
        /// This client certificate was used by the device provisioning service to register the enrollment with IoT Hub.
```

File: provisioning/device/src/DeviceRegistrationResult.cs [hunk 2]
```diff
@@ -200,9 +200,9 @@ public class DeviceRegistrationResult
         public string JsonPayload { get; private set; }
 
         /// <summary>
-        /// The PEM encoded operational client certificate that was returned by the certificate authority.
+        /// The PEM encoded operational client certificate that was signed by the certificate authority.
         /// This operational certificate was used by the device provisioning service to register the enrollment with IoT Hub.
-        /// The IoT device can use this operational certificate to authenticate with IoT Hub.
+        /// The IoT device can then use this returned operational certificate along with the private key information to authenticate with IoT Hub.
```
Comment (timtay-microsoft) [inline provisioning/device/src/DeviceRegistrationResult.cs:205]:
```suggestion
        /// The IoT device can then use this returned client certificate along with the private key information to authenticate with IoT Hub.
```

File: provisioning/device/src/ProvisioningRegistrationAdditionalData.cs [hunk 3]
```diff
@@ -15,11 +15,11 @@ public class ProvisioningRegistrationAdditionalData
         public string JsonData { get; set; }
 
         /// <summary>
-        /// The PEM encoded operational client certificate request that the device provisioning service (DPS) will send to its linked certificate authority which will sign
+        /// The PEM encoded operational client certificate signing request that the device provisioning service (DPS) will send to its linked certificate authority which will sign
         /// and return an X509 device identity operational client certificate to the device.
```
Comment (timtay-microsoft) [inline provisioning/device/src/ProvisioningRegistrationAdditionalData.cs:19]:
```suggestion
        /// and return an X509 device identity client certificate to the device.
```

File: provisioning/device/src/ProvisioningRegistrationAdditionalData.cs [hunk 4]
```diff
@@ -15,11 +15,11 @@ public class ProvisioningRegistrationAdditionalData
...
-        /// DPS will register the device and operational client certificate thumbprint in IoT Hub and return the certificate to the IoT device.
-        /// The IoT device can then use the operational certificate to authenticate with IoT Hub.
+        /// DPS will register the device and operational client certificate thumbprint in IoT Hub and return the certificate with the public key to the IoT device.
```
Comment (timtay-microsoft) [inline provisioning/device/src/ProvisioningRegistrationAdditionalData.cs:20]:
```suggestion
        /// DPS will register the device and client certificate thumbprint in IoT Hub and return the certificate with the public key to the IoT device.
```

File: provisioning/device/src/ProvisioningRegistrationAdditionalData.cs [hunk 5]
```diff
@@ -15,11 +15,11 @@ public class ProvisioningRegistrationAdditionalData
...
+        /// The IoT device can then use the returned operational certificate along with the private key information to authenticate with IoT Hub.
```
Comment (timtay-microsoft) [inline provisioning/device/src/ProvisioningRegistrationAdditionalData.cs:21]:
```suggestion
        /// The IoT device can then use the returned certificate along with the private key information to authenticate with IoT Hub.
```

**Review by timtay-microsoft — APPROVED** [review]
"Operational" is kind of a vague term to use in our documentation. Otherwise, looks good!

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**